processor/memorylimiter/memorylimiter.go

```diff
@@ -238,44 +239,89 @@ func (ml *memoryLimiter) forcingDrop() bool {
 	return atomic.LoadInt64(&ml.forceDrop) != 0
 }
 
-func (ml *memoryLimiter) memCheck() {
-	ms := ml.readMemStats()
-	ml.memLimiting(ms)
+func (ml *memoryLimiter) setForcingDrop(b bool) {
+	var i int64
+	if b {
+		i = 1
+	} else {
+		i = 0
+	}
```

Comment:
The else branch can be removed from here.

Reply:
done.